### ISSUE TITLE
Is the pin `urllib3 ~= 1.26.9` in ncbi-dataset-pylib intentional?

### ISSUE THREAD

**Issue by corneliusroemer** [ncbi/datasets#228]
urllib3 verson 2.0.0 was recently released.

conda-forge now uses that in its build system. But the `ncbi-dataset-pylib` pypi packagehas a pin `urllib3 ~= 1.26.9` which causes the conda-forge's recipe to fail with `pip check`.

I'll add the pin to the bioconda recipe, but was still curious whether that pin was intentional.

The whole requirements seem to be (from  the tar ball: https://files.pythonhosted.org/packages/3b/f7/8217f6cb4d2328ccd11791a19b668ad98861839d5b231f676905b086faa4/ncbi-datasets-pylib-14.26.0.tar.gz)

```txt
# File base.txt
gffutils ~= 0.10.1
jinja2 ~= 3.1.2
jsonlines ~= 3.0.0
protobuf ~= 3.20.1
python-dateutil ~= 2.8.2
urllib3 ~= 1.26.9
python_dateutil >= 2.5.3

setuptools >= 21.0.0

urllib3 >= 1.25.3
```

Note that urllib3 is listed twice, not sure which part comes from the OpenAPI generator and which is from your source code: `urllib3 >= 1.25.3`

See PR that made me notice this here: https://github.com/bioconda/bioconda-recipes/commit/3102bbba3aa19754bea1fc58dc3a73f233c1ecdf

**Comment by BradHolmes** [comment]
Hi Cornelius,

Thank you for bringing this to our attention.  We'll need to investigate the right option here, as I no longer know which urllib3 pinning is needed either.  

Could you point me to where the conda-forge recipe is failing with `pip check`?

**Comment by corneliusroemer** [comment]
Thanks @BradHolmes 

It failed here:
https://github.com/bioconda/bioconda-recipes/runs/13186808109
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=32686&view=logs&jobId=e14e69ff-a0ae-55c4-b71d-229b239cfb2f

I've fixed by making conda deps align with pip.

**Comment by BradHolmes** [comment]
If I understand correctly, the [bioconda recipe](https://github.com/bioconda/bioconda-recipes/blame/master/recipes/ncbi-datasets-pylib/meta.yaml#L21), at line 21, doesn't peg `pip`.  While we don't maintain the bioconda recipe, we could make a PR to get that pinned properly, or if you have one already, we could support that one.  

Meanwhile, our [conda build](https://github.com/conda-forge/ncbi-datasets-cli-feedstock) is doing ok.  It doesn't do anything with python.  Instead, it publishes our Datasets command line application.

How would you like to proceed @corneliusroemer ?

**Comment by corneliusroemer** [comment]
@BradHolmes I think there's a misunderstanding. I've fixed the conda recipe, it doesn't require any action from you. The issue isn't that pip isn't pinned but that your pip package has tighter requirements than the way the conda recipe was written. The conda recipe is just how I noticed this issue, it's not the core point of _this_ issue here in this repo.

The fundamental question is whether you intended that pin in this package, especially the fact that there are two lines for the same package in the requirements.txt looked odd so I thought I'll ask. Does that make sense now?
